proxy_http.go

```diff
@@ -399,6 +405,13 @@ func copyHeadersForForwarding(dst, src http.Header) {
 			}
 		}
 	}
+
+	pc := src.Get("Proxy-Connection")
+	if pc != "" {
+		log.Debugf("Got Proxy-Connection header from %#v", src)
+		dst.Set("Connection", pc)
+	}
+	dst.Del("Proxy-Connection")
```

Comment:
This is the heart of the change. Instead of doing anything special for the case where the proxy _is not chained_, just by default assume it's a 1-hop proxy and properly handle Proxy-Connection.

Note that this requires a corresponding change in flashlight.